packages/common-all/src/types/events.ts

```diff
@@ -0,0 +1,68 @@
+import { Disposable } from "./compat";
+import { EventEmitter as NodeEventEmitter } from "events";
+
+/**
+ * Represents a typed event that can be subscribed to.
+ *
+ * This has an indentical interface signature to vscode.Event, but can be used
+ * outside of plugin-core.
+ */
+export interface Event<T> {
+  /**
+   * A function that represents an event to which you subscribe by calling it with
+   * a listener function as argument.
+   *
+   * @param listener The listener function will be called when the event happens.
+   * @param thisArgs The `this`-argument which will be used when calling the event listener.
+   */
+  (listener: (e: T) => any, thisArgs?: any): Disposable;
+}
+
+/**
+ * An event emitter can be used to create and manage an {@link Event} for others
+ * to subscribe to. One emitter always owns one event.
+ *
+ * This mimics the interface signature of vscode.EventEmitter but can be used
+ * outside of plugin-core.
+ */
+export class EventEmitter<T> {
+  _emitter: NodeEventEmitter = new NodeEventEmitter();
```

Comment:
is this available in the browser? if not, should it be moved to `common-server`?

Reply:
Is there a definitive way to tell during compile time (or do we just get a runtime error if running in browser with a non-browser supported function)?  I was thinking that our `tsconfig.json` build setup should catch this if it were a problem, but I wasn't sure.

Reply:
tried webpack locally, it seems to have worked - so I think we should be ok?  Let me know if you think there are more solid ways to test this out.

Reply:
lgtm